src/ImageIO/CGImageSource.cs

```diff
@@ -376,5 +376,42 @@ public CGImageSourceStatus GetStatus (int index)
 		{
 			return CGImageSourceGetStatusAtIndex (handle, index);
 		}
+
+		[iOS(11,0)][Mac (10,13)][Watch(4,0)][TV(11,0)]
+		[DllImport (Constants.ImageIOLibrary)]
+		static extern IntPtr /* CFDictionaryRef* */ CGImageSourceCopyAuxiliaryDataInfoAtIndex (IntPtr /* CGImageSourceRef* */ isrc, nuint index, IntPtr /* CFStringRef* */ auxiliaryImageDataType);
+
+		public CGImageAuxiliaryDataInfo CopyAuxiliaryDataInfo (CGImageSource imageSource, nuint index, CGImageAuxiliaryDataType auxiliaryImageDataType)
+		{
+			var ptr = CGImageSourceCopyAuxiliaryDataInfoAtIndex (imageSource.Handle, index, auxiliaryImageDataType.GetConstant ().Handle);
+			if (ptr == IntPtr.Zero)
+				return null;
+			
+			var dictionary = new NSDictionary (ptr);
```

Comment:
It's not clear if the same `NSDictionary` can be returned between calls and it's better to avoid multiple managed instance over the same native one. Use `Runtime.GetNSObject<NSDictionary> (ptr)` to reuse an existing instance (if any) or create a new one.

That `dictionary` is local to the method and should be disposed (ideally inside a `using` clause)